### PR TITLE
Support search in multi-select field

### PR DESCRIPTION
**Pull request by shraddhaSankpal27**

### CODE REVIEW

**Review by bharatkashyap — APPROVED** [review]
Looks good to me

File: projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.ts [hunk 1]
```diff
@@ -33,8 +33,10 @@ export class DynamicMultipleDropdownComponent implements OnInit, OnChanges, OnDe
 
   _: any = _;
 
-
   public isDependsInvalid: any;
+  public isSearchables: any;
+  public showMaterSelect: boolean = true;
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.ts:38]: can we make it `isSearchables` as `isSearchable` and also spell check `showMaterSelect`

File: projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html [hunk 2]
```diff
@@ -4,17 +4,28 @@
...
+              <span>
+                {{optionValueToOptionLabelMap.get(fromJS(value))}}
+              </span>
+              <i (click)="removeItem(optionValueToOptionLabelMap.get(fromJS(value)))" class="delete icon"></i>
+              <span *ngIf="index < formControlRef.value?.length - 1 && !_.isEmpty(optionValueToOptionLabelMap.get(fromJS(value)))">
               </span>
             </span>
           </ng-container>
+          <ng-container>
+            <span *ngIf="isSearchables">
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html:21]: same `isSearchables` here.

File: projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html [hunk 3]
```diff
@@ -54,10 +65,9 @@
         </div>
       </div>
     </div>
-
     <div class="sb-modal-dropdown-web" *ngIf="platform === 'web'"  [hidden]="!showModal" >
       <ul>
-        <li (click)="checkUncheckAll()">
+        <li (click)="checkUncheckAll()" [hidden]="!showMaterSelect">
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html:70]: spell check `showMaterSelect`

**Review by venkateshwarans — COMMENTED** [review]
check comments

File: projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.ts (hunk 1 above)
Comment (shraddhaSankpal27) [reply projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.ts]: Done

File: projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html (hunk 2 above)
Comment (shraddhaSankpal27) [reply projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html]: Done

File: projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html (hunk 3 above)
Comment (shraddhaSankpal27) [reply projects/common-form-elements/src/lib/dynamic-multiple-dropdown/dynamic-multiple-dropdown.component.html]: Done

**Comment by pallakartheekreddy** [comment]
@bharatkashyap As discussed, Please validate with old from config with depends property. Please validate backwards compatibility as well.

File: projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ export class DynamicDropdownComponent implements OnInit, OnChanges, OnDestroy {
   @Input() isMultiple?: boolean;
   @Input() context?: FormControl;
   @Input() contextTerms?: any;
-  @Input() formControlRef?: CustomFormControl;
+  @Input() formControlRef?: FormControl;
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts:24]: `CustomFormControl` is needed. can you please revert this change?

Comment (shraddhaSankpal27) [reply projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts]: Done

**Review by bharatkashyap — APPROVED** [review]
LGTM

File: projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts [hunk 5]
```diff
@@ -65,8 +73,12 @@ export class DynamicDropdownComponent implements OnInit, OnChanges, OnDestroy {
       this.options = [];
     }
 
-    if (!_.isEmpty(this.field.sourceCategory)) {
-      this.formControlRef.sourceCategory = this.field.sourceCategory;
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts:69]: This line is needed.

**Review by venkateshwarans — COMMENTED** [review]
Check comment

File: projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts (hunk 5 above)
Comment (shraddhaSankpal27) [reply projects/common-form-elements/src/lib/dynamic-dropdown/dynamic-dropdown.component.ts]: reverted code change

**Review by pallakartheekreddy — APPROVED** [review]
